drivers/input/input_mcux_kpp.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2025, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/device.h>
+#include <fsl_kpp.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/dt-bindings/clock/imx_ccm.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/sys/util.h>
+
+LOG_MODULE_REGISTER(kpp, CONFIG_INPUT_LOG_LEVEL);
+
+#define DT_DRV_COMPAT nxp_mcux_kpp
+
+#define INPUT_KPP_COLUMNNUM_MAX  KPP_KEYPAD_COLUMNNUM_MAX
+#define INPUT_KPP_ROWNUM_MAX     KPP_KEYPAD_ROWNUM_MAX
+
+struct kpp_config {
+	KPP_Type *base;
+	const struct device *ccm_dev;
+	clock_control_subsys_t clk_sub_sys;
+	const struct pinctrl_dev_config *pcfg;
+};
+
+
+struct kpp_data {
+	struct k_mutex lock;
+};
+
+static void get_source_clk_rate(const struct device *dev, uint32_t *clk_rate)
+{
+	const struct kpp_config *dev_cfg = dev->config;
+	const struct device *ccm_dev = dev_cfg->ccm_dev;
+	clock_control_subsys_t clk_sub_sys = dev_cfg->clk_sub_sys;
+	uint32_t rate = 0;
+
+	if (device_is_ready(ccm_dev)) {
+		clock_control_get_rate(ccm_dev, clk_sub_sys, &rate);
+	} else {
+		LOG_ERR("CCM driver is not installed");
+		*clk_rate = rate;
+		return;
+	}
+	*clk_rate = rate;
+}
+
+static void input_kpp_scan(const struct device *dev, uint8_t *data)
+{
+	const struct kpp_config *config = dev->config;
+	struct kpp_data *drv_data = dev->data;
+	uint32_t clk_rate;
+
+	k_mutex_lock(&drv_data->lock, K_FOREVER);
+
+	get_source_clk_rate(dev, &clk_rate);
+
+	KPP_keyPressScanning(config->base, data, clk_rate);
+
+	k_mutex_unlock(&drv_data->lock);
+}
+
+static void input_kpp_data_analyze(const struct device *dev, uint8_t *data)
+{
+	const struct kpp_config *config = dev->config;
+	struct kpp_data *drv_data = dev->data;
+	int row_data = 0;
+	int col_data = 0;
+	int row_data_mask;
+
+	k_mutex_lock(&drv_data->lock, K_FOREVER);
+
+	for (int col = 0; col < INPUT_KPP_COLUMNNUM_MAX; col++) {
+		if (data[col] != 0) {
+			row_data_mask = data[col];
+			for	(int row = 0; row < INPUT_KPP_ROWNUM_MAX; row++) {
+				if ((row_data_mask & (1 << row)) == 0) {
+					row_data = row;
+					col_data = col;
+					continue;
+				}
+			}
+		}
+	}
+	input_report_abs(dev, INPUT_ABS_X, col_data, false, K_FOREVER);
+	input_report_abs(dev, INPUT_ABS_Y, row_data, false, K_FOREVER);
+	KPP_SetSynchronizeChain(config->base, kKPP_ClearKeyDepressSyncChain);
+	input_report_key(dev, INPUT_BTN_TOUCH, 1, true, K_FOREVER);
+
+	k_mutex_unlock(&drv_data->lock);
+}
+
+static void kpp_isr(const struct device *dev)
+{
+	k_busy_wait(2000);
+
+	const struct kpp_config *config = dev->config;
+
+	uint8_t read_keys[INPUT_KPP_COLUMNNUM_MAX] = {0};
```

Comment:
does this need to be initialized? if `KPP_keyPressScanning` is meant to override the whole array you may as well just leave it uninitilized